frontend/src/components/waitlist/WaitlistPage.tsx

```diff
@@ -42,14 +42,19 @@ export const WaitlistPage = (props: Props) => {
   }, [email, usersData]);
 
   useEffect(() => {
-    setLocale(
-      props.supportedLocales.find(
-        (supportedLocale) =>
-          supportedLocale.toLowerCase() ===
-          currentLocale.split("-")[0].toLowerCase(),
-      ) ?? "en",
-    );
-  }, [currentLocale, props.supportedLocales]);
+    // Check if navigator and navigator.language are available
+    if (navigator && navigator.language) {
+      const userLanguage = navigator.language.toLowerCase();
```

Comment:
Looking at the docs, it [looks like](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language) `navigator.language` would also be something like `pl-PL`, so just a strict match would not match that - we'll want to match on the language part.

```suggestion
      const userLanguage = navigator.language.split("-")[0].toLowerCase();
```